src/tests/src/solver/expressions/test_AstDOTStyleVisitor.cpp

```diff
@@ -0,0 +1,153 @@
+legend_VariableNode [ label =" VariableNode: 1"]
+legend_MultiplicationNode [ label =" MultiplicationNode: 1"]
+legend_NegationNode [ label =" NegationNode: 1"]
+legend_ParameterNode [ label =" ParameterNode: 1"]
+legend_PortFieldNode [ label =" PortFieldNode: 1"]
+legend_ComponentParameterNode -> legend_ComponentVariableNode [style=invis];
+legend_ComponentVariableNode -> legend_DivisionNode [style=invis];
+legend_DivisionNode -> legend_EqualNode [style=invis];
+legend_EqualNode -> legend_GreaterThanOrEqualNode [style=invis];
+legend_GreaterThanOrEqualNode -> legend_LessThanOrEqualNode [style=invis];
+legend_LessThanOrEqualNode -> legend_LiteralNode [style=invis];
+legend_LiteralNode -> legend_MultiplicationNode [style=invis];
+legend_MultiplicationNode -> legend_NegationNode [style=invis];
+legend_NegationNode -> legend_ParameterNode [style=invis];
+legend_ParameterNode -> legend_PortFieldNode [style=invis];
+legend_PortFieldNode -> legend_SubtractionNode [style=invis];
+legend_SubtractionNode -> legend_SumNode [style=invis];
+legend_SumNode -> legend_VariableNode [style=invis];
+}
+}
+)raw";
+
+    Node* literalNode = registry.create<LiteralNode>(-40.);
+    Node* negationNode = registry.create<NegationNode>(literalNode);
+    Node* parameterNode = registry.create<ParameterNode>("avogadro_constant");
+    Node* multiplicationNode = registry.create<MultiplicationNode>(negationNode, parameterNode);
+    Node* variableNode = registry.create<VariableNode>("atoms_count");
+    Node* divisionNode = registry.create<DivisionNode>(variableNode, multiplicationNode);
+    Node* portFieldNode = registry.create<PortFieldNode>("gasStation", "1149");
+    Node* sumNode = registry.create<SumNode>(divisionNode, portFieldNode);
+    Node* componentVariableNode = registry.create<ComponentVariableNode>("1150", "otherStation");
+    Node* componentParameterNode = registry.create<ComponentParameterNode>("1151", "otherConstant");
+    Node* subtractionNode = registry.create<SubtractionNode>(componentVariableNode,
+                                                             componentParameterNode);
+    Node* equalNode = registry.create<EqualNode>(subtractionNode, sumNode);
+    Node* literalNode2 = registry.create<LiteralNode>(53.);
+    Node* lessThanOrEqualNode = registry.create<LessThanOrEqualNode>(literalNode2, equalNode);
+    Node* literalNode3 = registry.create<LiteralNode>(54.);
+    Node* greaterThanOrEqualNode = registry.create<GreaterThanOrEqualNode>(literalNode3,
+                                                                           lessThanOrEqualNode);
+
+    return std::pair{expected_msg, greaterThanOrEqualNode};
+}
+
+BOOST_FIXTURE_TEST_CASE(tree_with_all_type_node, Registry<Node>)
+{
+    auto [expected_msg, expr] = buildAllTypeNode(*this);
+
+    
+    std::ostringstream os;
+
+    AstDOTStyleVisitor astGraphVisitor;
+    
+    std::pair<AstDOTStyleVisitor&, Node*> pair1(astGraphVisitor, expr);
+    os << pair1;
```

Comment:
I know that @a-zakir and @flomnes discussed on what follows and made a decision, but I'm still going to leave of comment about it : 
It's about : 
```C++
AstDOTStyleVisitor astGraphVisitor;
std::pair<AstDOTStyleVisitor&, Node*> pair1(astGraphVisitor, expr);
*os << pair1;
```
The use of this visitor seems not very friendly to me ; it feels we do strange things because we can't do more simple.
Indeed :  
- First defining a **std::pair** of 2 different objects feels unusual and then give it to a stream is not very friendly.
- We put the result in a **std::ostream** and not in a **std::string** (which seems more immediate)
- as we test a visitor, and more precisely the only operation it has (the dispatch / visiting), we could expect something like : 
  ```C++
  AstDOTStyleVisitor astGraphVisitor(...);
  std::string dot_content = astGraphVisitor.dispatch(expr);
  ```
  as the dot content is the result of the visit / dispatch operation.

Reply:
We could define a wrapper such as
```cpp
std::string exprToDOT(Node* expr)
{
    AstDOTStyleVisitor astGraphVisitor;
    std::ostringstream os;
    os << std::pair(astGraphVisitor, expr);
    return os.str();
}

Reply:
i am totaly ok with your remark @guilpier-code  the oveloard of << operator is ugly